org.eclipse.scout.rt.jackson/src/main/java/org/eclipse/scout/rt/jackson/dataobject/DataObjectDeserializer.java

```diff
@@ -48,8 +50,22 @@ protected IDataObject deserializeDataObject(JsonParser p, DeserializationContext
         return p.getCodec().readValue(p, IDoEntity.class); // delegate to DoEntityDeserializer for object-like structure
       case START_ARRAY:
         return p.getCodec().readValue(p, DoList.class); // delegate to DoCollectionDeserializer for collection-like structure (using DoList as generic structure instead of DoSet or DoCollection)
+      case VALUE_STRING:
+        return createSimpleValueObject(p.getValueAsString());
+      case VALUE_NUMBER_INT:
+        return createSimpleValueObject(p.getValueAsLong());
+      case VALUE_NUMBER_FLOAT:
+        return createSimpleValueObject(p.getValueAsDouble());
```

Comment:
Other code locatoins transform `VALUE_JSON_FLOAT` without knowing the Java target type into a `BigDecimal`:
```suggestion
        return createSimpleValueObject(p.getCodec().readValue(p, BigDecimal.class));
```